src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -9,6 +9,21 @@ public class RegistrationServiceImpl implements RegistrationService {
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new InvalidUserException("User must not be null");
+        }
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new InvalidUserException("A user with this login is already registered");
+        }
+        if (user.getLogin() == null || user.getLogin().length() < 6) {
+            throw new InvalidUserException("Login must contain at least 6 characters");
```

Comment:
Split the condition checking `user.getLogin()` into separate `if` statements to provide more specific error messages. For example, one `if` could check for `null` and another for the length being less than 6. This will help the user understand exactly what is wrong with the input data. Refer to the checklist for the bad and good example.